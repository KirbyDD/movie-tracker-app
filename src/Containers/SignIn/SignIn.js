import React, { Component } from 'react';
import { connect } from 'react-redux';
import { signIn } from '../../actions';
import { fetchUser } from '../../apiCalls'


class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      currentUser: '',
      error: ''
    }
  }

  handleInput = e => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleUserSignIn = (e) => {
    e.preventDefault()
    fetchUser({email: this.state.email, password: this.state.password})
      .then(response => {
        if (response.data === undefined) {
          this.setState({error: 'Incorrect Email/Password'})
        } else {
        this.props.signUserIn(response.data)}})
      .then(res => this.handleResetState())
      .catch(error => this.setState({error: error.message}))
  }

  handleResetState = () => {
    this.setState({
      email: '',
      password: '',
      currentuser: '',
      error: ''
    })
  }
  render() {
    return (
      <form >
        <label htmlFor="signIn-email">Email</label>
        <input 
          type="email" 
          placeholder="Enter Email Here" 
          name="email" 
          value={this.state.email} 
          id="signIn-email" 
          onChange={(e) => this.handleInput(e)}/>
        <label htmlFor="signIn-password">Password</label>
        <input
          type="text"
          placeholder-="Enter Password Here"
          name="password"
          value={this.state.password}
          id="signIn-password"
          onChange={(e) => this.handleInput(e)}/>
        <button
        onClick={(e) => this.handleUserSignIn(e)}>Sign In</button>
      </form>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  signUserIn: (user) => dispatch( signIn(user) )
})

export default connect(null, mapDispatchToProps)(SignIn);