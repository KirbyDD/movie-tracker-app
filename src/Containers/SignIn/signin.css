* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}
.signin-form {
  height: 500px;
  width: 450px;
  border-radius: 7px;
  margin-left: auto;
  margin-right: auto;
  background-image: linear-gradient(rgba(0, 0, 0, 0.822), rgba(0, 0, 0, 0.733));
  box-shadow: 0 0 10px 5px rgba(255, 255, 255, 0.027);
  margin-top: 120px;
  display: flex;
  flex-direction: column;
}

.signin-container {
  border: 1px solid transparent;
  width: 100vw;
  height: 100vh;
  background-image: linear-gradient(rgba(0, 0, 0, 0.274), rgba(0, 0, 0, 0.5));
}


input {
  width: 100%;
  height: 40px;
  padding-left: 10px;
  font-size: 20px;
  border: none;
  border-radius: 5px;
  display: inline-block;
  outline: none;
}

label {
  color: #fff;
  display: inline-block;
}

.input-container {
  height: 70px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
  width: 80%;
  margin-top: 10px
}

.signin-btn {
  height: 40px;
  width: 80%;
  border-radius: 5px;
  background: rgba(1, 210, 120, 0.753);
  cursor: pointer;
  font-size: 16px;
  outline:none;
}

.signin-btn:hover {
  background: #01D277;
}

.input-btn-container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 300px;
  margin-top: 40px;
}

.sigin-header {
  color: #fff;
  margin-top: 50px;
}

.error-msg {
  color: red;
  height: 20px;
}