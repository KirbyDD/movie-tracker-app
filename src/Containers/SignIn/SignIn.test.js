import React from 'react'
import {SignIn, mapDispatchToProps} from './SignIn'
import { shallow } from 'enzyme'

describe('SignIn Container', () => {
  describe.only('SignIn Component', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<SignIn />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should hold the email, password, and error in local state', () => {
      const expected = {
        email: '',
        password: '',
        error: null
      }
      const result = wrapper.state()
      expect(result).toEqual(expected)
    })

    it('should set the state as the input changes', () => {
      let event = {target: {name: 'email', value: 'ryan@flachman.net'}}
      let expected = {email: 'ryan@flachman.net', password: '', error: null}
      wrapper.instance().handleInput(event)
      expect(wrapper.state()).toEqual(expected)
    })

    it('should be able to reset the state of the email and password', () => {
      wrapper.instance().setState({email: 'ryan@flachman.net'})
      wrapper.instance().setState({password: 'password'})
      const expectedBeforeReset = {email: 'ryan@flachman.net', password: 'password', error: null}
      expect(wrapper.state()).toEqual(expectedBeforeReset)
      wrapper.instance().handleResetInputs()
      const expectedAfterResest = {email: '', password: '', error: null}
      expect(wrapper.state()).toEqual(expectedAfterResest)
      
    })

    it('Should be able to reset the error property to null', () => {
      wrapper.instance().setState({error: 'This is a mock error message'})
      const expectedBeforeReset = 'This is a mock error message'
      expect(wrapper.state().error).toEqual(expectedBeforeReset)
      wrapper.instance().handleResetError()
      const expectedAfterResest = null
      expect(wrapper.state().error).toEqual(expectedAfterResest)
    })

    it('Should be able to set the error property to Incorrect Email/Password', () => {
      const expectedErrorState = null
      expect(wrapper.state().error).toEqual(expectedErrorState)
      wrapper.instance().handleSettingError()
      const exptectedErrorStateNow = 'Incorrect Email/Password'
      expect(wrapper.state().error).toEqual(exptectedErrorStateNow)
    })
  })
})