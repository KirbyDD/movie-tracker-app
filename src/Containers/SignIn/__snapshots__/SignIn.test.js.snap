// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Container SignIn Component should match the snapshot 1`] = `
<div
  className="signin-container"
>
  <form
    className="signin-form"
  >
    <h2
      className="sigin-header"
    >
      Sign In
    </h2>
    <div
      className="input-btn-container"
    >
      <div
        className="input-container"
      >
        <label
          htmlFor="signIn-email"
        >
          Email
        </label>
        <input
          id="signIn-email"
          name="email"
          onChange={[Function]}
          type="email"
          value=""
        />
      </div>
      <div
        className="input-container"
      >
        <label
          htmlFor="signIn-password"
        >
          Password
        </label>
        <input
          id="signIn-password"
          name="password"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <p
        className="error-msg"
      />
      <button
        className="signin-btn"
        onClick={[Function]}
      >
        Sign In
      </button>
    </div>
  </form>
</div>
`;
